packages/react/src/xcss-prop/__tests__/media-query-strict.test.tsx

```diff
@@ -0,0 +1,179 @@
+/** @jsxImportSource @compiled/react */
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { cssMap as cssMapLoose } from '@compiled/react';
+import { render } from '@testing-library/react';
+
+import { cssMap } from '../../create-strict-api/__tests__/__fixtures__/strict-api';
+import type { XCSSProp } from '../../create-strict-api/__tests__/__fixtures__/strict-api';
+import type { XCSSAllProperties, XCSSAllPseudos } from '../index';
+
+function CSSPropComponent({
+  testId,
+  xcss,
+}: {
+  testId?: string;
+  xcss: ReturnType<typeof XCSSProp<XCSSAllProperties, XCSSAllPseudos>>;
+}) {
+  return (
+    <div data-testid={testId} className={xcss}>
+      foo
+    </div>
+  );
+}
+
+const styles = cssMap({
+  invalidMediaObject: {
+    // @ts-expect-error — @media at rule object is not allowed in strict cssMap
+    '@media': {
+      screen: { color: 'red' },
+    },
+  },
+  invalidMediaQuery: {
+    // @ts-expect-error — this specific @media is not in our allowed types
+    '@media (min-width: 100px)': {
+      color: 'red',
+    },
+  },
+  valid: {
```

Comment:
I renamed this to `validMediaQueryInvalidProperty` and added a _valid_ scenario, but there's no difference as `valid` and `validMediaQueryInvalidProperty` are both not erroring, eg. `validMediaQueryInvalidProperty` would error _here_, but not in the `xcss={styles.validMediaQueryInvalidProperty}`.